backup/queries_statistics.go

```diff
@@ -119,14 +119,15 @@ func GetAttributeStatistics(connectionPool *dbconn.DBConn, tables []Table) map[u
 		inheritClause, statSlotClause, statCollationClause,
 		SchemaFilterClause("n"), utils.SliceToQuotedString(tablenames))
 
-	results := make([]AttributeStatistic, 0)
-	err := connectionPool.Select(&results, query)
+	rows, err := connectionPool.Query(query)
 	gplog.FatalOnError(err)
-	stats := make(map[uint32][]AttributeStatistic)
-	for _, stat := range results {
-		stats[stat.Oid] = append(stats[stat.Oid], stat)
+	for rows.Next() {
+		var attStat AttributeStatistic
+		err = rows.StructScan(&attStat)
+		gplog.FatalOnError(err)
+		processRow(&attStat)
 	}
-	return stats
+	gplog.FatalOnError(rows.Err())
```

Comment:
Isn't this check excessive? There is already an error check after `StructScan`.

Reply:
after StructScan, the Next method may return false, but the error can only be checked using the Err method